server/text_generation_server/utils/logits_process.py

```diff
@@ -498,9 +498,10 @@ def __call__(
     ):
         if fsm_grammar_state == -1 or self.fsm is None:
             return logits
-        allowed_tokens = self.fsm.allowed_token_ids(fsm_grammar_state)
+        allowed_tokens = self.fsm.get_next_instruction(fsm_grammar_state).tokens
         mask = torch.full_like(logits, -math.inf)
-        mask[:, allowed_tokens] = 0
+        if allowed_tokens is not None:
+	    mask[:, allowed_tokens] = 0
```

Comment:
it looks like lines 503 and 504 are on the same indented block. maybe indentation is needed?

```suggestion
        if allowed_tokens is not None:
	        mask[:, allowed_tokens] = 0
```

Reply:
just figured out the first indent is a tab, not 4 spaces.

![image](https://github.com/user-attachments/assets/2a7452da-de14-406e-964d-26349fd464f1)